src/main/io/gps.c

```diff
@@ -434,17 +434,17 @@ static void ubloxSendDataUpdateChecksum(const uint8_t *data, uint8_t len, uint8_
     }
 }
 
-static void ubloxSendMessage(const uint8_t *data, uint8_t len)
+static void ubloxSendMessage(const ubx_message *message, uint8_t len)
 {
     uint8_t checksumA = 0, checksumB = 0;
-    serialWrite(gpsPort, data[0]);
-    serialWrite(gpsPort, data[1]);
-    ubloxSendDataUpdateChecksum(&data[2], len - 2, &checksumA, &checksumB);
+    serialWrite(gpsPort, message->header.preamble1);
+    serialWrite(gpsPort, message->header.preamble2);
+    ubloxSendDataUpdateChecksum(&message->header.msg_class, len - 2, &checksumA, &checksumB);
```

Comment:
Now that you've ubx_message as first parameter, you can avoid passing the len parameter and use message->len+4 instead when asking for the checksum. (checksum includes all the payload plus class, ID and length itself, so 4 bytes)

Reply:
Updated. Can you test it with GPS module? I don't have any. Thank you.